presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayElementBlockWriter.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import io.airlift.slice.Slice;
+
+public class ArrayElementBlockWriter
+        extends AbstractArrayElementBlock
+        implements BlockBuilder
+{
+    private final BlockBuilder blockBuilder;
+    private int positionsWritten;
+    private boolean ongoingBlockEntry;
+
+    public ArrayElementBlockWriter(BlockBuilder blockBuilder, int start)
+    {
+        super(start);
+        this.blockBuilder = blockBuilder;
+    }
+
+    @Override
+    protected BlockBuilder getBlock()
+    {
+        return blockBuilder;
+    }
+
+    private void checkOngoingBlockEntry()
+    {
+        if (ongoingBlockEntry) {
```

Comment:
this is never set to true, so I don't think this check is working. Also add a test to catch that